app/controllers/admins_controller.rb

```diff
@@ -0,0 +1,141 @@
+# Insight Service Approval APIs
+#
+# This is a API to query approval services
+#
+# OpenAPI spec version: 1.0.0
+#
+# Generated by: https://github.com/swagger-api/swagger-codegen.git
+#
+class AdminsController < ApplicationController
+  def add_action
+    group = Group.find(params[:group_id])
+    action = group.actions.create!(action_params)
+
+    json_response(action, :created)
+  end
+
+  def add_group
+    group = Group.create!(group_params)
+    json_response(group, :created)
+  end
+
+  def add_request
+    workflow = Workflow.find(params[:workflow_id])
+    req = workflow.requests.create!(request_params)
+
+    json_response(req, :created)
+  end
+
+  def add_template
+    template = Template.create!(template_params)
+    json_response(template, :created)
+  end
+
+  def add_workflow
+    workflow = Template.find(params[:template_id]).workflows.create!(workflow_params)
+    json_response(workflow, :created)
+  end
+
+  def remove_action
+    group = Group.find(params[:group_id])
+    action = group.actions.find_by!(id: params[:id]) if group
```

Comment:
what if this line fails? should you use `Group.find!` above? otherwise `action.destroy` may end up with `nil.destroy`

Reply:
`Group.find` will raise an `ActiveRecord::RecordNotFound` exception if not found, `find_by` will return `nil`, so use `find_by!`. In normal case, `action` should always be available when `action.destroy` is called.